src/actions/orders/create.ts

```diff
@@ -0,0 +1,107 @@
+import type { FastifyReply, FastifyRequest } from "fastify";
+import { Order, OrderStatus } from "../../models/Order";
+import { Product } from "../../models/Product";
+import { InvalidAmountError } from "../../errors/invalidAmount";
+import { NotFoundError } from "objection";
+import { OrderItem } from "../../models";
+
+type Request = FastifyRequest<{
+    Body: {
+        customer_id: number;
+        items: {
+            product_id: number;
+            quantity: number;
+            discount?: number;
+        }[];
+    }
+}>;
+
+export default async (
+    { body: { customer_id, items } }: Request,
+    reply: FastifyReply
+) => {
+    try {
+        const orderItems = items ?? []
+        const products = await Product.query().findByIds(orderItems.map(item => item.product_id));
+        const missingProduct = orderItems.filter(item => !products.find(product => product.id === item.product_id));
+
+        if (missingProduct.length) {
+            throw new NotFoundError({ message: 'Product not found' + missingProduct.map(item => item.product_id).join(',') });
+        }
+
+        const productPriceMap = new Map(products.map(product => [product.id, product.price]));
+
+        let total_paid = 0;
```

Comment:
Here too, we can create a `src/services/orderService.ts` and a `calculateOrderTotals` function